tests/testthat/test-top-factors-getter.R

```diff
@@ -0,0 +1,89 @@
+# These tests check that the getTopFactors getter function works correctly
+# 
+# Tests include:
+# 1. Checking that the top factors order in getTopFactors matches that in 
+#    getOutDf
+# 2. Checking that the number of factors to output can be specified
+# 3. Checking that the weights are correctly ordered
+# 4. Checking that the top factors and weights are stable (always have the same
+#    values for a fixed data set and seed)
+
+context("Checking top factors getter works")
+
+set.seed(256)
+
+# Build training data
```

Comment:
I got a warning message: `glm.fit: fitted probabilities numerically 0 or 1 occurred ` when I ran the prep. I assume that's because it's synthetic data. And the output is captured so it doesn't matter in devtools::test